File: l-2-3/ts/models/TextContainer.ts
import { TextRow } from "./TextRow";

export class TextContainer {
    private rows: TextRow[] = [];

    addRow(row: TextRow): void {
        this.rows.push(row);
    }

    removeRow(index: number): void {
        if (index >= 0 && index < this.rows.length) {
            this.rows.splice(index, 1);
        }
    }

    removeRowsContaining(substring: string): void {
        this.rows = this.rows.filter(row => !row.getContent().includes(substring));
    }

    clear(): void {
        this.rows = [];
    }

    getLongestRowLength(): number {
        if (this.rows.length === 0) return 0;
        return Math.max(...this.rows.map(row => row.getLength()));
    }

    capitalizeAllRows(): void {
        for (const row of this.rows) {
            row.capitalizeWords();
        }
    }

    getAllRows(): string[] {
        return this.rows.map(row => row.getContent());
    }
}

File: l-2-3/cs/models/TextContainer.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class TextContainer
{
    private List<TextRow> rows = new List<TextRow>();

    public void AddRow(TextRow row)
    {
        rows.Add(row);
    }

    public void RemoveRow(int index)
    {
        if (index >= 0 && index < rows.Count)
            rows.RemoveAt(index);
    }

    public void RemoveRowsContaining(string substring)
    {
        rows = rows.Where(row => !row.GetContent().Contains(substring)).ToList();
    }

    public void Clear()
    {
        rows.Clear();
    }

    public int GetLongestRowLength()
    {
        if (rows.Count == 0) return 0;
        return rows.Max(row => row.GetLength());
    }

    public void CapitalizeAllRows()
    {
        foreach (var row in rows)
        {
            row.CapitalizeWords();
        }
    }

    public List<string> GetAllRows()
    {
        return rows.Select(row => row.GetContent()).ToList();
    }
}
